File: LiveReload/Classes/Plugins/ToolOutput.m

#import "ToolOutput.h"

@implementation ToolOutput

@synthesize compiler=_compiler;
@synthesize project=_project;
@synthesize type=_type;
@synthesize sourcePath=_sourcePath;
@synthesize line=_line;
@synthesize message=_message;
@synthesize output=_output;

- (instancetype)initWithCompiler:(Compiler *)compiler type:(ToolOutputType)type sourcePath:(NSString *)sourcePath line:(NSInteger)line message:(NSString *)message output:(NSString *)output {
    self = [super init];
    if (self) {
        _compiler = [compiler retain];
        _type = type;
        _sourcePath = [sourcePath copy];
        _line = line;
        _message = [message copy];
        _output = [output copy];
    }
    return self;
}

@end

File: LiveReload/Classes/Plugins/ToolOutput.swift
import Foundation
import LRActionKit

public enum ToolOutputType{
    case Log
    case Error
    case ErrorRaw
}

public class ToolOutput: NSObject {

    public let action: Action
    public let type: ToolOutputType
    public let sourcePath: String?
    public let line: Int?
    public let message: String?
    public let output: String?

    public init(action: Action, type: ToolOutputType, sourcePath: String?, line: Int?, message: String?, output: String?) {
        self.action = action
        self.type = type
        self.sourcePath = sourcePath
        self.line = line
        self.message = message
        self.output = output
        super.init()
    }
    
}
